#
# glibc-2.2.5 "thread exit" fix
# (Category - Recommended bug fix)
# (Source - glibc cvs glibc-2-2-branch)
#
# http://sources.redhat.com/ml/libc-hacker/2002-02/msg00119.html
# http://www.zipworld.com.au/%7Egschafer/patches/glibc-2.2.5.thread-exit.patch.gz
#
diff -ru glibc-2.2.5-orig/linuxthreads/signals.c glibc-2.2.5/linuxthreads/signals.c
--- glibc-2.2.5-orig/linuxthreads/signals.c	2002-01-21 04:21:16.000000000 +0100
+++ glibc-2.2.5/linuxthreads/signals.c	2003-09-13 18:01:50.000000000 +0200
@@ -198,7 +198,7 @@
         s != __pthread_sig_cancel &&
         s != __pthread_sig_debug) {
       sigdelset(&mask, s);
-      if (sighandler[s].old == NULL ||
+      if (sighandler[s].old == (arch_sighandler_t) SIG_ERR ||
           sighandler[s].old == (arch_sighandler_t) SIG_DFL ||
           sighandler[s].old == (arch_sighandler_t) SIG_IGN) {
         sa.sa_handler = pthread_null_sighandler;
